server/temporal-client/with-auth-metadata.js

```diff
@@ -25,23 +25,27 @@ const getter = (obj, property) => {
   return obj[property];
 };
 
-const extractAccessToken = (ctx) => {
+const extractTokens = (ctx) => {
   if (ctx.isAuthenticated()) {
-    return ctx.state.user.accessToken;
+    const { accessToken, idToken } = ctx.state.user;
+    return { accessToken, idToken };
   }
-  return undefined;
+  return { accessToken: undefined, idToken: undefined };
 };
 
 const buildGrpcMetadata = async (ctx) => {
   const metadata = new grpc.Metadata();
 
   const auth = await getAuthConfig();
   if (auth.enabled) {
-    const accessToken = extractAccessToken(ctx);
+    const { accessToken, idToken } = extractTokens(ctx);
     if (!accessToken) {
-      throw Error('Request unauthorized')
+      throw Error('Request unauthorized');
     }
     metadata.add('authorization', `Bearer ${accessToken}`);
+    if (idToken) {
```

Comment:
Shouldn't this rather be `if (passIdToken)`?

Reply:
There are two primary parts of auth flow:
1. Read auth config. When user authenticates, store auth data in session (`server/auth/index.js`)
2. When making requests to server, check if session contains user auth data. Add it if it exists  (`server/temporal-client/with-auth-metadata.js)

`with-auth-metadata` (#2) is only responsible for reading auth data from session and adding it to server request. It actually doesn't know what particular provider configuration was used 

`server/auth` (#1) on the other hand works with particular auth provider configs and is responsible to act according to auth configuration

Lmk if this answers the question